### PR TITLE
Fix CMake build on Windows

### PR DESCRIPTION
**Pull request by NlightNFotis**
Attempting to build `cbmc` with `cmake` on `Visual Studio 2019/Win 10` fails. 

This PR introduces some needed changes to two `CMakeLists.txt` files to make
it work.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 project(CBMC)
 
-find_package(BISON)
-find_package(FLEX)
+find_package(BISON REQUIRED)
```
Comment (hannes-steffenhagen-diffblue) [inline src/CMakeLists.txt:3]: 👍 it's a non-optional dependency and absolutely should be marked as `REQUIRED`

File: jbmc/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ add_subdirectory(unit)
 
 add_custom_target(java-models-library ALL
     COMMAND mvn --quiet -Dmaven.test.skip=true package
-    COMMAND cp target/core-models.jar ${CMAKE_CURRENT_SOURCE_DIR}/src/java_bytecode/library/
+    COMMAND ${CMAKE_COMMAND} -E copy target/core-models.jar ${CMAKE_CURRENT_SOURCE_DIR}/src/java_bytecode/library/
```
Comment (hannes-steffenhagen-diffblue) [inline jbmc/CMakeLists.txt:7]: Yep, we should avoid using unix-y commands in our... commands.

I'd recommend putting all the variable stuff in `"` quotes as well... I don't actually know off the top of my head how CMake handles variable expansions with whitespace in them.

File: jbmc/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ add_subdirectory(unit)
 
 add_custom_target(java-models-library ALL
     COMMAND mvn --quiet -Dmaven.test.skip=true package
```
Comment (hannes-steffenhagen-diffblue) [inline jbmc/CMakeLists.txt:6]: Not an action for this PR, but we should also avoid hardcoding the maven executable. By adding a configuration for this we could support cases where people want to build with `~/.local/bin/maven-experimental` or whatever.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

File: jbmc/CMakeLists.txt (hunk 2 above)
Comment (chrisr-diffblue) [reply jbmc/CMakeLists.txt:7]: It's no worse than the _current_ situation so I'm happy for this to be merged as is - but It probably is worth trying a build on a checkout of CBMC that has spaces in its file path. On Mac/Windows it's not _that_ uncommon for this to happen.

**Review by chrisr-diffblue — APPROVED** [review]

**Pull request merged**